java/core/src/main/java/com/microsoft/bond/PrimitiveBondType.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond;
+
+import java.util.HashMap;
+
+/**
+ * Partially implements the {@link BondType} contract for primitive data types.
+ * @param <TPrimitive> the class of the primitive value
+ */
+public abstract class PrimitiveBondType<TPrimitive> extends BondType<TPrimitive> {
+
+    // restrict subclasses to the current package
+    PrimitiveBondType(Class<? extends PrimitiveBondType<TPrimitive>> thisClass) {
+        super(thisClass.hashCode());
+    }
+
+    @Override
+    public final boolean isNullableType() {
+        return false;
+    }
+
+    @Override
+    public final boolean isGenericType() {
+        return false;
+    }
+
+    @Override
+    public final BondType<?>[] getGenericTypeArguments() {
+        return null;
+    }
+
+    @Override
+    final boolean equalsInternal(BondType<?> obj) {
+        // the caller makes sure that the class of the argument is the same as the class of this object,
+        // and since primitive types are singletons all instance are considered equal
+        return true;
```

Comment:
Please go back to the public equals method that calls this private method and walk through what it does and/or read the comments. Two instances of the same primitive Bond Type are considered equal, as they represent the same Bond type.

---
In reply to: [123115913](https://github.com/Microsoft/bond/pull/502#discussion_r123115913) [](ancestors = 123115913)